app/services/price_policy_updater.rb

```diff
@@ -39,10 +44,47 @@ def assign_attributes
 
   def save
     ActiveRecord::Base.transaction do
-      @price_policies.all?(&:save) || raise(ActiveRecord::Rollback)
+      (assign_attributes && @price_policies.all?(&:save)) || raise(ActiveRecord::Rollback)
+    end
+  end
+
+  def assign_attributes_for_stepped_billing
```

Comment:
We should consider how to refactor this in the next sprint

Reply:
Yes, 100%!